backend/src/API/shoutoutAPI.ts

```diff
@@ -36,3 +27,14 @@ export const getShoutouts = async (
   }
   return ShoutoutsDao.getShoutouts(memberEmail, type);
 };
+
+export const hideShoutout = async (body: Shoutout, user: IdolMember): Promise<Shoutout> => {
+  const canEdit = await PermissionsManager.canHideShoutouts(user);
+  if (!canEdit) {
+    throw new PermissionError(
+      `User with email: ${user.email} does not have permission to hide shoutouts!`
+    );
+  }
+  await ShoutoutsDao.hideShoutout(body);
+  return body;
```

Comment:
It seems like the `hideShoutout` handler is just updating the shoutout document with whatever is provided from the frontend. I think it makes more sense for this handler to just take the uuid of the shoutout that we want to hide and then update the document with the the new hidden field in the dao layer. e.g. 

ShoutoutAPI.ts
```
const hideShoutout = (uuid, user) => {
    ...
    await ShoutoutsDao.hideShoutout(uuid)
```
ShoutoutsDao.ts
```
static async hideShoutout(uuid) {
    ...
    await shoutousCollection.doc(uuid).update({ hidden: true })
```

Reply:
I was originally planning to do this, but do you think there is any use case for updating other fields of a Shoutout? Thus, `hideShoutout` could be converted to a function like `updateShoutout` and won't just update the `hidden` field.

Reply:
yeah that's a good point -- maybe instead the dao layer can just be an update operation that takes in the updated shoutout object. but in the API layer i think the function should still just take in just a uuid and pass in the update to the dao layer

Reply:
Would this require us to have a getShoutout(uuid) function in the dao layer to check if the shoutout exists? I'm just unclear as to how having the uuid as the only parameter into the API layer will allow updated information to get to the dao layer

Reply:
You can check whether the shoutout exists by:
```
const doc = await shoutoutsCollection.doc(uuid).get();
```
If doc is `null`, then it doesnt exist. The reason we only need to the `uuid` in the API layer is because we just want to update the `hidden` field of the shoutout to be `true`, so we don't need to pass in the whole shoutout object.

Reply:
@henry-li-06 For some reason passing the uuid is causing a "type error somewhere in the api-dao interaction, and the error is causing the api to fail with the internal server error" according to @JacksonStaniec. Jackson and I discussed this and we think it's best to leave the current implementation so we can use shoutouts tomorrow during all-hands, but revisit this issue after.

Reply:
I was referring to the gateway-api interaction. The gateway layer should just expecting the request body to be in the form of 
```
{
  uuid: string
}
```
Then the gateway layer should pass `req.body.uuid` to the API layer. The business logic in the API layer can look something like this: 
```
const hideShoutout = (uuid: string, user: IdolMember) => {
  // check permissions 
  // get the shoutout with uuid
  // check that shoutout exists 
  // create updated shoutout that's hidden 
  // call updateShoutout in the dao layer to update the shoutout 
}
```

Reply:
It seems like the reason there was a type error is here:
![image](https://user-images.githubusercontent.com/65922473/201441756-51f0977b-bdef-414c-b518-a421f9ada228.png)
The frontend is passing `uuid` as the entire request body instead of `{ uuid }` so the gateway layer can't parse the `uuid` field from the request body.

Reply:
@henry-li-06 hmm that still doesn't work for me and @JacksonStaniec